internal/common/backoff/retrypolicy.go

```diff
@@ -136,27 +130,27 @@ func (p *ExponentialRetryPolicy) SetMaximumAttempts(maximumAttempts int) {
 }
 
 // ComputeNextDelay returns the next delay interval.  This is used by Retrier to delay calling the operation again
-func (p *ExponentialRetryPolicy) ComputeNextDelay(elapsedTime time.Duration, numAttempts int) time.Duration {
+func (p *ExponentialRetryPolicy) ComputeNextDelay(elapsedTime time.Duration, attempt int) time.Duration {
 	// Check to see if we ran out of maximum number of attempts
-	if p.maximumAttempts != noMaximumAttempts && numAttempts >= p.maximumAttempts {
+	if p.maximumAttempts != retry.NoMaximumAttempts && attempt >= p.maximumAttempts {
```

Comment:
Should these be IgnoreMaximumAttempts?

Reply:
I think the name is okay as it is, it's also consistent with other similar names in this package.

Reply:
Let's maybe call it `UnlimitedMaximumAttempts`?